lib/currency.js

```diff
@@ -0,0 +1,196 @@
+const { Amount, pkg: hsdPkg } = require('hsd');
+const assert = require('bsert');
+
+const { isChainSupported } = require('./chain');
+
+export const CURRENCY_UNITS = {
+  bitcoin: {
+    currency: 'bitcoin',
+    unit: 'btc',
+    milli: 'mbtc',
+    micro: 'bit',
```

Comment:
I figured that file should mirror handshake's `pkg` (https://github.com/handshake-org/hsd/blob/master/lib/pkg.js) which only has `unit`, `base`, and `currency`. My guess is this was deliberate since other denominations tend to have more flexible nomenclature based on the community (as in the [Bitcoin Wiki](https://en.bitcoin.it/wiki/Units)). Base and currency/unit have to be a little stricter since the rest are based on that. That's why in handshake code for `Amount` they use this same convention for the intermediate currency labels:
```
    milli: `m${hsdPkg.unit}`,
    micro: `u${hsdPkg.unit}`,
```
I think there's an argument to be made for having each major unit "hard coded" in though. Maybe more useful as an open discussion on either one of those projects directly?

Reply:
I don't want to hold up this PR with too much, maybe we could talk about this more but also I don't think its the end of the world hard coding in `bits`

Reply:
Cool sounds good. Like I said, probably still worth mentioning on the bcoin PR to see if that's something we want hardcoded in pkg.